File: codeforces/Python/639div2c.py
t = int(input())
for _ in range(t):
    n = int(input())
    als = list(map(int, input().split()))
    ans = "YES"
    ls = [0 for _ in range(n)]
    for i in range(n):
        if ls[(als[i]+i)%n] != 0:
            ans = "NO"
        else:
            ls[(als[i]+i)%n] += 1
    print(ans)

File: codeforces/cpp/639div2c.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

int main() {
    ll t;
    cin >> t;

    for (int tt=0; tt<t; tt++) {
        ll n;
        cin >> n;

        vector<ll> av(n, 0);
        ll a;
        string ans = "YES";
        for (int i=0; i<n; i++) {
            cin >> a;
            if (av.at((a+i)%n)) {
                ans = "NO";
            } else {
                av.at((a+i)%n)++;
            }
        }
        cout << ans << endl;

    }
}